### PR TITLE
remove fopen_s and fix error in logger

### PR DESCRIPTION
**Pull request by lplewa**
remove usage of fopen_s and fix error message in the logger.
check commit descriptions for details. 

### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files

### CODE REVIEW

File: test/utils/utils_log.cpp [hunk 1]
```diff
@@ -175,13 +175,14 @@ TEST_F(test, parseEnv) {
...
                             if (logLevel.second <= LOG_INFO) {
                                 expect_fput_count = 1;
-                            } else {
-                                expect_fput_count = 0;
                             }
                         } else {
                             expect_fput_count = 1;
-                            expect_fopen_count = 0;
-                            expected_stream = stderr;
+                            if (expected_filename.size() > 256) {
```
Comment (bratpiorka) [inline test/utils/utils_log.cpp:181]: could we use here MAX_FILE_PATH?

File: src/utils/utils_log.c [hunk 2]
```diff
@@ -156,26 +156,23 @@ void util_log_init(void) {
...
-#ifdef _WIN32
-            if (fopen_s(&loggerConfig.output, file, "w")) {
-                loggerConfig.output = NULL;
-            }
-#else
-            loggerConfig.output = fopen(file, "w");
-#endif
+        if (len > MAX_FILE_PATH) {
+            loggerConfig.output = stderr;
+            LOG_ERR("Cannot open output file - path too long", file);
```
Comment (kswiecicki) [inline src/utils/utils_log.c:161]: There's no string format specifier `%s` to log the `file` variable.

File: test/utils/utils_log.cpp (hunk 1 above)
Comment (lplewa) [reply test/utils/utils_log.cpp]: Done

File: src/utils/utils_log.c (hunk 2 above)
Comment (lplewa) [reply src/utils/utils_log.c]: Done

**Review by PatKamin — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**